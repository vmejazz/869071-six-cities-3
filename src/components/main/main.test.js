import React from "react";
import renderer from "react-test-renderer";
// import {Main} from "./main.jsx";
jest.mock(`../map/map.jsx`, () => `Map`);
jest.mock(`../city-list/city-list.jsx`, () => `CityList`);

// const OFFER_PLACES = 254;

// const offersArray = [
//   {
//     id: 1,
//     title: ``,
//     price: 120,
//     srcImg: ``,
//     srcGallery: [],
//     description: ``,
//     premium: true,
//     type: ``,
//     rate: 1,
//     bedrooms: 1,
//     maxGuests: 1,
//     apartmentStuff: [],
//     ownerInfo: {
//       name: ``,
//       super: true,
//       srcAvatar: ``
//     },
//     position: [1, 1],
//     city: ``
//   }
// ];

// const userInfo = {
//   AuthorizationStatus: `NO_AUTH`
// };

// const cityes = {
//   AMSTERDAM: [52.38333, 4.9]
// };

it(`<Main /> should render all page`, () => {
  const tree = renderer
      .create(
          <h1></h1>
          // <Main
          //   activeCity={`Moscow`}
          //   offerPlacesCount={OFFER_PLACES}
          //   offersArray={offersArray}
          //   cityes={cityes}
          //   onApartmentCardClick={() => {}}
          //   userInfo={userInfo}
          // />
      )
      .toJSON();

  expect(tree).toMatchSnapshot();
});
