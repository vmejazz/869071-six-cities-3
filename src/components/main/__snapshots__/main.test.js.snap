// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render all page 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index + "
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <CityList
          activeCity="Moscow"
          cityes={
            Object {
              "AMSTERDAM": Array [
                52.38333,
                4.9,
              ],
            }
          }
        />
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            254
             places to stay in 
            Moscow
          </b>
          <form
            action="#"
            className="places__sorting"
            method="get"
          >
            <span
              className="places__sorting-caption"
            >
              Sort by
            </span>
            <span
              className="places__sorting-type"
              tabIndex={0}
            >
              Popular
              <svg
                className="places__sorting-arrow"
                height={4}
                width={7}
                xmlns="http://www.w3.org/2000/svg"
              >
                <use
                  xlinkHref="#icon-arrow-select"
                  xmlnsXlink="http://www.w3.org/1999/xlink"
                />
              </svg>
            </span>
            <ul
              className="places__options places__options--custom places__options--opened"
            >
              <li
                className="places__option"
                onClick={[Function]}
                tabIndex={0}
              >
                Popular
              </li>
              <li
                className="places__option"
                onClick={[Function]}
                tabIndex={0}
              >
                Price: low to high
              </li>
              <li
                className="places__option"
                onClick={[Function]}
                tabIndex={0}
              >
                Price: high to low
              </li>
              <li
                className="places__option"
                onClick={[Function]}
                tabIndex={0}
              >
                Top rated first
              </li>
            </ul>
          </form>
          <div
            className="cities__places-list places__list tabs__content"
          />
        </section>
        <div
          className="cities__right-section"
        >
          <Map
            cityes={
              Object {
                "AMSTERDAM": Array [
                  52.38333,
                  4.9,
                ],
              }
            }
            offersShow={Array []}
          />
        </div>
      </div>
    </div>
  </main>
</div>
`;
